Having two io registers without a direction causes an assertion

To reproduce:

Slave.scala:


```scala
import Chisel._

class Slave extends Module {
  val io = new Bundle {
    val send = Reg(UInt(0, 8))
    val recv = Reg(UInt(0, 8))
  }
}

class SlaveTests(c: Slave) extends Tester(c) {

}

object hello {
  def main(args: Array[String]): Unit = {
    chiselMainTest(args.slice(1, args.length),
      () => Module(new Slave())){c => new SlaveTests(c)}
  }
}
```

```
csguest@cosi6:~/brainfuck_cpu/hw/spi$ sbt
Picked up JAVA_TOOL_OPTIONS: -javaagent:/usr/share/java/jayatanaag.jar 
[info] Set current project to spi (in build file:/home/csguest/brainfuck_cpu/hw/spi/)
> run Slave --backend c --compile --test --genHarness
[info] Compiling 1 Scala source to /home/csguest/brainfuck_cpu/hw/spi/target/scala-2.11/classes...
[warn] there were 9 feature warnings; re-run with -feature for details
[warn] one warning found
[info] Running brainfuck_cpu.spi.hello Slave --backend c --compile --test --genHarness
CPP elaborate
[info] [0.049] // COMPILING < (class brainfuck_cpu.spi.Slave)>(0)
[info] [0.054] giving names
[error] (run-main-0) java.lang.AssertionError: assertion failed
java.lang.AssertionError: assertion failed
	at scala.Predef$.assert(Predef.scala:151)
	at Chisel.Backend$NameSpace$2.reserveName(Backend.scala:147)
	at Chisel.Backend$$anonfun$nameAll$1$$anonfun$apply$11.apply(Backend.scala:182)
	at Chisel.Backend$$anonfun$nameAll$1$$anonfun$apply$11.apply(Backend.scala:182)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at Chisel.Backend$$anonfun$nameAll$1.apply(Backend.scala:182)
	at Chisel.Backend$$anonfun$nameAll$1.apply(Backend.scala:161)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at Chisel.Backend.nameAll(Backend.scala:161)
	at Chisel.Backend.elaborate(Backend.scala:838)
	at Chisel.CppBackend.elaborate(Cpp.scala:1510)
	at Chisel.Driver$.execute(Driver.scala:94)
	at Chisel.Driver$.apply(Driver.scala:41)
	at Chisel.Driver$.apply(Driver.scala:53)
	at Chisel.chiselMain$.apply(hcl.scala:66)
	at Chisel.chiselMainTest$.apply(hcl.scala:79)
	at brainfuck_cpu.spi.hello$.main(Slave.scala:33)
	at brainfuck_cpu.spi.hello.main(Slave.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
[trace] Stack trace suppressed: run last compile:run for the full output.
java.lang.RuntimeException: Nonzero exit code: 1
	at scala.sys.package$.error(package.scala:27)
[trace] Stack trace suppressed: run last compile:run for the full output.
[error] (compile:run) Nonzero exit code: 1
[error] Total time: 4 s, completed Jul 11, 2015 2:22:30 PM
```

--- a/src/main/scala/Backend.scala
+++ b/src/main/scala/Backend.scala
@@ -178,7 +178,8 @@ abstract class Backend extends FileSystemUtilities{
           } yield new_cand).head // only use the first one
         } else candidate
       }
-      def reserveName(name: String): Unit = assert(name == getUniqueName(name))
+      // Ignore attempts to reserve an empty ("") name - pr499, issue 459
+      def reserveName(name: String): Unit = if (name != "") assert(name == getUniqueName(name), "name " + name + " cannot be reserved")
       def getUniqueName(candidate: String): String = {
         val unique_name = ensureUnique(candidate)
         namespace += unique_name.toLowerCase

--- a/src/test/scala/NameTest.scala
+++ b/src/test/scala/NameTest.scala
@@ -479,6 +479,7 @@ class NameSuite extends TestSuite {
   /* XXX test case derived from issue #6 on github.
    */
   @Test def testInputPortNameChange() {
+    println("testInputPortNameChange:")
     class InputPortNameComp extends Module {
       val io = new Bundle {
         val in = Bits(INPUT, 20)
@@ -498,6 +499,7 @@ class NameSuite extends TestSuite {
   /* XXX test case derived from issue #153 on github.
    */
   @Test def testNameItTooEager153() {
+    println("testNameItTooEager153:")
     class MyBundle extends Bundle
     {
       val x = Bool()
@@ -539,5 +541,25 @@ class NameSuite extends TestSuite {
         "--targetDir", dir.getPath.toString()),
         () => Module(new NameItTooEager153()))
     
+  }
+
+  /* Multiple directionless IO's don't throw assertion error - issue #459.
+   */
+  @Test def testMultipleDirectionlessIO459() {
+    println("testMultipleDirectionlessIO459:")
+    class MultipleDirectionlessIO459 extends Module {
+      val io = new Bundle {
+        val send = Reg(UInt(0, 8))
+        val recv = Reg(UInt(0, 8))
+      }
+    }
+    
+    // This should fail since we don't assign a directiom to the IO ports.
+    intercept[IllegalStateException] {
+      chiselMain(Array[String]("--backend", "v",
+          "--targetDir", dir.getPath.toString()),
+          () => Module(new MultipleDirectionlessIO459()))
+    }
+    assertTrue(ChiselError.hasErrors)
   }
 }

